lib/datasource/pod/index.ts

```diff
@@ -95,29 +117,53 @@ async function requestGithub<T = unknown>(
 }
 
 const githubRegex = regEx(
-  /^https:\/\/github\.com\/(?<account>[^/]+)\/(?<repo>[^/]+?)(\.git|\/.*)?$/
+  /(?<hostURL>(^https:\/\/[a-zA-z0-9-.]+))\/(?<account>[^/]+)\/(?<repo>[^/]+?)(\.git|\/.*)?$/
```

Comment:
this would now also match `https://gitea.com/org/repo.git`, so it can probably fail for existing users now?

Reply:
To confirm this locally I added the below tests:

```
    it('returns null for 404 Github 2', async () => {
      httpMock
        .scope('https://gitea.com')
        .get('/api/v3/repos/org/repo/contents/Specs/a/c/b/foo')
        .reply(404)
        .get('/api/v3/repos/org/repo/contents/a/c/b/foo')
        .reply(404)
        .get('/api/v3/repos/org/repo/contents/Specs/foo')
        .reply(404)
        .get('/api/v3/repos/org/repo/contents/foo')
        .reply(404);
      const res = await getPkgReleases({
        ...config,
        registryUrls: [
          ...config.registryUrls,
          'https://gitea.com/org/repo',
        ],
      });
      expect(res).toBeNull();
    });



    it('processes real data from Github2 with shard with specs', async () => {
      httpMock
        .scope('https://gitea.com')
        .get('/api/v3/repos/org/repo/contents/Specs/a/c/b/foo')
        .reply(200, [{ name: '1.2.3' }]);
      const res = await getPkgReleases({
        ...config,
        registryUrls: ['https://gitea.com/org/repo'],
      });
      expect(res).toEqual({
        registryUrl: 'https://gitea.com/org/repo',
        releases: [
          {
            version: '1.2.3',
          },
        ],
      });
    });
    it('processes real data from Github2 with shard without specs', async () => {
      httpMock
        .scope('https://gitea.com')
        .get('/api/v3/repos/org/repo/contents/Specs/a/c/b/foo')
        .reply(404)
        .get('/api/v3/repos/org/repo/contents/a/c/b/foo')
        .reply(200, [{ name: '1.2.3' }]);
      const res = await getPkgReleases({
        ...config,
        registryUrls: ['https://gitea.com/org/repo'],
      });
      expect(res).toEqual({
        registryUrl: 'https://gitea.com/org/repo',
        releases: [
          {
            version: '1.2.3',
          },
        ],
      });
    });
    it('processes real data from Github2 with specs without shard', async () => {
      httpMock
        .scope('https://gitea.com')
        .get('/api/v3/repos/org/repo/contents/Specs/a/c/b/foo')
        .reply(404)
        .get('/api/v3/repos/org/repo/contents/a/c/b/foo')
        .reply(404)
        .get('/api/v3/repos/org/repo/contents/Specs/foo')
        .reply(200, [{ name: '1.2.3' }]);
      const res = await getPkgReleases({
        ...config,
        registryUrls: ['https://gitea.com/org/repo'],
      });
      expect(res).toEqual({
        registryUrl: 'https://gitea.com/org/repo',
        releases: [
          {
            version: '1.2.3',
          },
        ],
      });
    });
    it('processes real data from Github2 without specs without shard', async () => {
      httpMock
        .scope('https://gitea.com')
        .get('/api/v3/repos/org/repo/contents/Specs/a/c/b/foo')
        .reply(404)
        .get('/api/v3/repos/org/repo/contents/a/c/b/foo')
        .reply(404)
        .get('/api/v3/repos/org/repo/contents/Specs/foo')
        .reply(404)
        .get('/api/v3/repos/org/repo/contents/foo')
        .reply(200, [{ name: '1.2.3' }]);
      const res = await getPkgReleases({
        ...config,
        registryUrls: ['https://gitea.com/org/repo'],
      });
      expect(res).toEqual({
        registryUrl: 'https://gitea.com/org/repo',
        releases: [
          {
            version: '1.2.3',
          },
        ],
      });
    });
  });

```

They all passed:
```
 PASS  lib/datasource/pod/index.spec.ts
  datasource/pod/index
    ✓ processes real data from Github Enterprise with shard with specs (4 ms)
    ✓ processes real data from Github Enterprise with shard without specs (9 ms)
    ✓ processes real data from Github Enterprise with specs without shard (10 ms)
    ✓ processes real data from Github Enterprise without specs without shard (15 ms)
    getReleases
      ✓ returns null for invalid inputs (24 ms)
      ✓ returns null for empty result (5 ms)
      ✓ returns null for 404 (18 ms)
      ✓ returns null for 404 Github enterprise (19 ms)
      ✓ returns null for 404 Github 2 (17 ms)
      ✓ returns null for 404 Github enterprise with different url style (15 ms)
      ✓ returns null for 401 (5 ms)
      ✓ throws for 429 (32 ms)
      ✓ returns null for unknown error (5 ms)
      ✓ processes real data from CDN (6 ms)
      ✓ processes real data from Github with shard with specs (5 ms)
      ✓ processes real data from Github with shard without specs (9 ms)
      ✓ processes real data from Github with specs without shard (12 ms)
      ✓ processes real data from Github without specs without shard (17 ms)
      ✓ processes real data from Github2 with shard with specs (5 ms)
      ✓ processes real data from Github2 with shard without specs (10 ms)
      ✓ processes real data from Github2 with specs without shard (13 ms)
      ✓ processes real data from Github2 without specs without shard (17 ms)

Test Suites: 1 passed, 1 total
Tests:       22 passed, 22 total
Snapshots:   0 total
Time:        3.498 s
Ran all test suites within paths "lib/datasource/pod/index.spec.ts".
```

Don't think we need to actually add these tests as we already have something similar.

Reply:
yes, but gitea isn't github, so it will fail on runtime.

I think we should do an `HEAD` reuest to the api root and check for github specific headers. So we only try further requests on gihub compatible api.

Reply:
If the host and path exist the `getReleasesFromGithub` method will be able to call the URL.

if the request passes it will return `releases`
else if the request fails it will return `null`

Isn't this sufficient?

Reply:
We tried parsing the header but unfortunately in the error case it does not return the response headers and I would think it would be out of the scope of this PR to change the request and error handling approach. The impact of this is that we make extra few requests.

Reply:
Fundamentally, the change here is isn't doing anything new: previously,  if the githubURL somehow matched it would make two requests, and the response and additional load would be 2 404'd requests.  With the new approach, if the githubURL that the user configures somehow matches, we'll now have 4 404'd requests.   We'd need to make 1 HTTP request  anyway just to check the headers.  

If a renovate user has configured the URL against gitea, it will still fail because it's not a github or github Enterprise server -- just as it does today.

Reply:
@viceice I am resolving this comment...please unresolve it if you don't agree.